pullrequests/flood_fill/step1.go

```diff
@@ -0,0 +1,24 @@
+//lint:file-ignore U1000 Ignore all unused code
+package floodfill
+
+/*
+時間：13分
+再帰を使えば良さそうというのはすぐにわかりましたが、自分にとっては実装が少し難しく感じたので、そこそこの時間がかかりました。
+ただ、これぐらいの難易度の問題をちゃんと解ききり、解いた後は脳内デバッグをしてバグを見つけて取り除いた後に1発で正解することができたので、これまでのLeetCodeの練習の成果が出ているなと感じることができました。
+コードについてもわりかし綺麗に書けたと思っていて、今回は特に修正すべきところを自分では見つけられませんでした。
+*/
+func floodFill(image [][]int, sr int, sc int, color int) [][]int {
+	originalColor := image[sr][sc]
+	if originalColor == color {
+		return image
+	}
+	image[sr][sc] = color
+	directions := [][]int{{0, 1}, {0, -1}, {1, 0}, {-1, 0}}
+	for _, direction := range directions {
+		newSr, newSc := sr+direction[0], sc+direction[1]
```

Comment:
`+` の前後にはスペースがあったほうが見やすそうです。gofmt を使うとスペースが入ると思うので、そちらが標準な気がします。（ドキュメントとかはとくに見つけられてないですが）

またこれは僕の推測ですが、`sr`, `sc` はそれぞれ `startRow`, `startColumn` の省略形なのかなと思ったので、シンプルに `row`, `column` でもよさそうです。（Golang の場合は `r`, `c` でも許容かもです）

Reply:
> `+` の前後にはスペースがあったほうが見やすそうです。gofmt を使うとスペースが入ると思うので、そちらが標準な気がします。（ドキュメントとかはとくに見つけられてないですが）

これはgofmtでフォーマット済みです。https://qiita.com/tchssk/items/77030b4271cd192d0347

> またこれは僕の推測ですが、sr, sc はそれぞれ startRow, startColumn の省略形なのかなと思ったので、シンプルに row, column でもよさそうです。

sr, scはLeetCodeで与えられている引数なので、そのまま使っていましたが、確かにもっと良い変数名にした方が良さそうですね。

> Golang の場合は r, c でも許容かもです

Go styleguideには下記のように書かれているため、Goの場合でもこの場合はr, cは許容されないと思います。

> Single-letter variable names can be a useful tool to minimize [repetition](https://google.github.io/styleguide/go/decisions#repetition), but can also make code needlessly opaque. Limit their use to instances where the full word is obvious and where it would be repetitive for it to appear in place of the single-letter variable.
https://google.github.io/styleguide/go/decisions#single-letter-variable-names